libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -244,8 +244,6 @@ internal class MapboxTripSession(
                         ) {
                             // workaround for a remaining issues in
                             // github.com/mapbox/mapbox-navigation-native/issues/3466
```

Comment:
I'm still able to reproduce https://github.com/mapbox/mapbox-navigation-android/issues/4727#issuecomment-926810538

> Noting that I'm lately missing the very first voice instruction and it seems definitely related to OP.

@SiarheiFedartsou @mskurydin @DmitryAk could you explain why the very first status is generating `null` `bannerInstruction` and `voiceInstruction`?

![first navigation status null banner and voice instructions](https://user-images.githubusercontent.com/1668582/135346290-3b354804-6ffe-4bb7-9550-e8705e7a1610.png)

It seems that `#3466` is not fully solved.

cc @S2Ler

Reply:
If I'm not mistaken, for `#3466` we need to switch to `setRouteAsync` function, which is what @korshaknn is doing in #4881. So in order to fully resolve issues with banner/voice instructions, we need to merge both this fix and #4881.

Reply:
That makes sense. Thanks for clarifying 🙇‍♂️

Reply:
We shouldn't really need async route setters to resolve missing voice instructions :thinking: 

If we are setting the route to NN only after the trip session starts, and all the observers are already subscribed when a  session starts, then we shouldn't miss the first voice event.

Reply:
Ah, I see you raised it in https://github.com/mapbox/mapbox-navigation-native/issues/3466#issuecomment-930539819 :+1: